addons/account/models/account_report.py

```diff
@@ -280,6 +280,17 @@ def _validate_formula(self):
                     expression.report_line_id.display_name,
                 ))
 
+    @api.constrains('sequence')
+    def _validate_account_report_line(self):
+        # by default the sequence is 0 for the account report lines.
+        sequence_with_zero = self.report_id.line_ids.filtered(lambda l: l.sequence == 0)
+        if len(sequence_with_zero) == 1:
+            parent_set = set()
```

Comment:
You could use a recordset and do |= to add elements into it here.

Reply:
Note that this changes the time complexity of the loop from O(n) to O(n^2) (which is now worse than the complexity of the sorting).
Probably only a detail, but nice to keep in mind :wink: